src/script/ceph-backport.sh

```diff
@@ -265,7 +266,12 @@ function prepare {
 
     debug "Cherry picking $number commits from ${github_endpoint}/pull/${original_pr} into local branch $local_branch"
     debug "If this operation does not succeed, you will need to resolve the conflicts manually"
-    git cherry-pick -x pr-$original_pr~$number..pr-$original_pr
+    let offset=$number-1
+    for ((i=$offset; i>=0; i--))
+    do
+        echo $i
+        git cherry-pick -x "pr-$original_pr~$i"
+    done
```

Comment:
I'd prefer to use the `debug` function instead of `echo` inside the loop. Perhaps something like the following would be acceptable?

```
for ((i=$number; i>0; i--))
do
    debug "Commits remaining to be cherry-picked: $i"
    git cherry-pick -x "pr-$original_pr~$i"
done
```

Reply:
:+1: for the debug line...that echo was actually a missed debug statement.

We'll still need `$offset` though as using number would cause an off-by-one error.

Reply:
> using number would cause an off-by-one error

How so? Notice that my proposed `for` statement is different from yours:

```
for ((i=$number; i>0; i--))
```

Reply:
I'd prefer for the loop to generate messages in English:

```
DEBUG: Commits remaining to be cherry-picked: 2
...
DEBUG: Commits remaining to be cherry-picked: 1
...
```

Hence my suggestion.

Reply:
Ah, we also need:

```
git cherry-pick -x "pr-$original_pr~(($i-1))i"
```